wandb/sdk/internal/sender.py

```diff
@@ -135,6 +133,19 @@ def _manifest_json_from_proto(manifest: "ArtifactManifest") -> Dict:
     }
 
 
+def _manifest_entry_from_proto(entry: "ArtifactManifestEntry") -> Dict:
+    birth_artifact_id = entry.birth_artifact_id if entry.birth_artifact_id else None
+    return {
+        "digest": entry.digest,
+        "birthArtifactID": birth_artifact_id,
+        "ref": entry.ref if entry.ref else None,
+        "size": entry.size if entry.size is not None else None,
+        "local_path": entry.local_path if entry.local_path else None,
+        "skip_cache": entry.skip_cache,
+        "extra": {extra.key: json.loads(extra.value_json) for extra in entry.extra},
+    }
```

Comment:
suggestion(optional): may be something you already ruled out, but is it worth using [`MessageToDict`](https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html) here (`from google.protobuf.json_format import MessageToDict`)?

to clarify a bit -- my thinking was something like this:
```suggestion
def _manifest_entry_from_proto(entry: "ArtifactManifestEntry") -> Dict:
    entry_dict = MessageToDict(entry, preserving_proto_field_name=True)
    entry_dict["birthArtifactID"] = entry_dict.pop("birth_artifact_id", None)
    entry_dict["extra"] = {
        extra["key"]: json.loads(extra["value_json"]) for extra in entry["extra"]
    }
    return entry_dict
```

Reply:
We could for sure, but I think at the moment I'll punt on this. Not a big lift, but kind of orthogonal to the thrust of this PR.